docs/ingress-gateway-external-lb/ingress-gateway.yaml

```diff
@@ -0,0 +1,124 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: asm-ingressgateway
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: asm-ingressgateway
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "watch", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: asm-ingressgateway
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: asm-ingressgateway
+subjects:
+- kind: ServiceAccount
+  name: asm-ingressgateway
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: asm-ingressgateway
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: asm-ingressgateway
+      istio: ingressgateway
+  template:
+    metadata:
+      annotations:
+        # This is required to tell Anthos Service Mesh to inject the gateway with the
+        # required configuration.
+        inject.istio.io/templates: gateway
+      labels:
+        app: asm-ingressgateway
+        istio: ingressgateway
+    spec:
+      containers:
+      - name: istio-proxy
+        image: auto # The image will automatically update each time the pod starts.
+        resources:
+          limits:
+            cpu: 2000m
+            memory: 1024Mi
+          requests:
+            cpu: 100m
+            memory: 128Mi
+      serviceAccountName: asm-ingressgateway
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: asm-ingressgateway
+  labels:
+    app: asm-ingressgateway
+    istio: ingressgateway
+spec:
+  ports:
+  # The status-port exposes a /healthz/ready endpoint that can be used for health checks.
+  - name: status-port
+    port: 15021
+    protocol: TCP
+    targetPort: 15021
+  - name: http2
+    port: 80
+  - name: https
+    port: 443    
+  selector:
+    app: asm-ingressgateway
+    istio: ingressgateway
+  type: NodePort # Using NodePort to expose the service on the nodes for the external load balancer.
+---
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: asm-ingressgateway
+spec:
+  maxUnavailable: 1
+  selector:
+    matchLabels:
+      app: asm-ingressgateway
+      istio: ingressgateway
+---
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: asm-ingressgateway
+spec:
+  maxReplicas: 5
+  metrics:
+  - type: Resource
+    resource:
+      name: cpu
+      target:
+        type: Utilization
+        averageUtilization: 80
+  minReplicas: 3
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: asm-ingressgateway
```

Comment:
Nitpick: Feel free to end this file with a new line?

Aside: This repo has an [.editorconfig](https://github.com/GoogleCloudPlatform/anthos-service-mesh-samples/blob/main/.editorconfig#L6) that automates this for you (but some editors require an [extension](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig)).

Reply:
Yes, my formatter usually does that but must have missed it for some reason. I have added it.